Jetchat/app/src/main/java/com/example/compose/jetchat/MainViewModel.kt

```diff
@@ -16,22 +16,16 @@
 
 package com.example.compose.jetchat
 
+import androidx.compose.material.DrawerState
+import androidx.compose.material.DrawerValue
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
 import androidx.lifecycle.ViewModel
-import kotlinx.coroutines.flow.MutableStateFlow
-import kotlinx.coroutines.flow.StateFlow
 
 /**
  * Used to communicate between screens.
  */
 class MainViewModel : ViewModel() {
-
-    private val _drawerShouldBeOpened = MutableStateFlow(false)
-    val drawerShouldBeOpened: StateFlow<Boolean> = _drawerShouldBeOpened
-
-    fun openDrawer() {
-        _drawerShouldBeOpened.value = true
-    }
-    fun resetOpenDrawerAction() {
-        _drawerShouldBeOpened.value = false
-    }
+    val drawerState by mutableStateOf(DrawerState(DrawerValue.Closed))
```

Comment:
@manuelvicnt is it a code smell to have a `DrawerState` inside a `ViewModel`?

Reply:
I returned the ViewModel to how it was before, since it wasn't essential to get the a11y changes to work.